### ISSUE TITLE
Question about weight initialization

### ISSUE THREAD

**Issue by yangwf1** [facebookresearch/maskrcnn-benchmark#817]
Hi, i'm using the distributed training. i have read the documents about torch.distributedDataParallel, we need to guarantee that model in each process should have weights with the same value. Although i checked that this condition is satisfied when running this code, however i haven't found any code that guarantees this, especially for those layers with random initialized weights. So i'm wondering how it is implemented.

**Comment by ppwwyyxx** [comment]
https://github.com/pytorch/pytorch/blob/81e70ffa19c90e14c20d1369aae86d19f305dc7d/torch/nn/parallel/distributed.py#L299

**Comment by fmassa** [comment]
As @ppwwyyxx mentioned, the weights are broadcasted in the construction of the DDP module, and then we assume that the optimizes behave the same on each process, and only distribute the gradients afterwards
